pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1963,12 +2080,38 @@ func (m *kubeGenericRuntimeManager) ListPodSandboxMetrics(ctx context.Context) (
 	return m.runtimeService.ListPodSandboxMetrics(ctx)
 }
 
-// isPodResizeInProgress checks whether the actuated resizable resources differ from the allocated resources
-// for any running containers. Specifically, the following differences are ignored:
+func (m *kubeGenericRuntimeManager) UpdateActuatedPodLevelResources(actuatedPod *v1.Pod) error {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		return nil
+	}
+
+	if actuatedPod.Spec.Resources == nil {
+		return nil
+	}
+
+	if actuatedPod.Spec.Resources.Requests == nil && actuatedPod.Spec.Resources.Limits == nil {
+		return nil
+	}
+
+	return m.actuatedState.SetPodLevelResources(actuatedPod.UID, actuatedPod.Spec.Resources)
+}
+
+// isPodResizeInProgress checks whether the actuated resizable resources differ from
+// the resources allocated for:
+// * any running containers - Specifically, the following differences are ignored:
 // - Non-resizable containers: non-restartable init containers, ephemeral containers
 // - Non-resizable resources: only CPU & memory are resizable
 // - Non-running containers: they will be sized correctly when (re)started
+// * any running pod if InPlacePodLevelResourcesVerticalScaling is enabled.
 func (m *kubeGenericRuntimeManager) IsPodResizeInProgress(allocatedPod *v1.Pod, podStatus *kubecontainer.PodStatus) bool {
+	if m.isContainerResourceResizeInProgress(allocatedPod, podStatus) {
+		return true
+	}
+
+	return m.isPodLevelResourcesResizeInProgress(allocatedPod, podStatus)
```

Comment:
optional nit, just a personal style preference


```suggestion
	return m.isContainerResourceResizeInProgress(allocatedPod, podStatus) || m.isPodLevelResourcesResizeInProgress(allocatedPod, podStatus)
```